File: 26.cpp
#include<iostream>
#include<conio.h>
#include<math.h>

using namespace std;

int main()
{
	int x,n,i,c;
	
	for(x=2;x<=1000;++x)
		{
			cout<<x<<" : ";
			c=x;
			while(2*(c/2)==c)
				c/=2;
			while(5*(c/5)==c)
				c/=5;
			cout<<c<<endl;
			continue;
			if(c==1)
				{
					cout<<"Non-recurring"<<endl;
					continue;
				}
			n=1;
			while(int(log10(c*n+1))!=log10(c*n+1))
				n++;
			cout<<n;
			cout<<endl;
			getch();
			
		}
}

File: 26.py
m=0
cm=0
for x in range(2,100001):
	print(x," : ",end="")
	c=x;
	while c%2==0:
		c/=2
	while c%5==0:
		c/=5
	if c==1:
		print("Non-recurring")
		continue
	n=1
	rem=1;
	while True:
		rem*=10%c
		rem%=c
		if(rem==1):
			break
		n+=1
	if n>m:
		m=n;
		cm=c;
	print(n)
	
	

print(cm,m)
